nevergrad/optimization/optimizerlib.py

```diff
@@ -860,6 +860,44 @@ def _internal_provide_recommendation(self) -> tp.ArrayLike:
         return self.avg
 
 
+class RescaleOptimizer(base.Optimizer):
+    """Proposes a version of a base optimizer which works at a different scale."""
+    def __init__(
+            self,
+            parametrization: IntOrParameter,
+            budget: tp.Optional[int] = None,
+            num_workers: int = 1,
+            base_optimizer: base.OptCls = CMA,
+            scale: tp.Optional[float] = None,
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        self.base_optimizer = base_optimizer(parametrization, budget=budget, num_workers=num_workers)
+        if scale is None:
+            assert budget is not None, "Either scale or budget must be known in RescaleOptimizer."
+            scale = np.sqrt(np.log(self.budget) / self.dimension)
+        self.scale = scale
+        assert self.scale != 0., "scale should be non-zero in Rescaler."
```

Comment:
stricly positive?

Reply:
I can see cases in which people might was to use -1. You can think of Opposite sampling.

Reply:
but would they want to *always* sample with a negative scale? that could work if you combine one positive and one negative I suppose, still feels a bit weird by why not

Reply:
I have in mind using Hammersley with scale 1 with half budget and Hammersley with scale -1 with the other half.
This is exactly Opposite sampling.
I think there is no drawback, and this might simplify some of our other codes in oneshot.py.